int8 updown
int8 leftright
int8 left_bump
int8 right_bump